vendor/github.com/hashicorp/go-retryablehttp/client.go

```diff
@@ -171,7 +171,7 @@ func DefaultRetryPolicy(resp *http.Response, err error) (bool, error) {
 	// the server time to recover, as 500's are typically not permanent
 	// errors and may relate to outages on the server side. This will catch
 	// invalid response codes as well, like 0 and 999.
-	if resp.StatusCode == 0 || resp.StatusCode >= 500 {
+	if resp.StatusCode == 0 || (resp.StatusCode >= 500 && resp.StatusCode != 501) {
```

Comment:
Did you mean to add this here?

Reply:
Yeah, I didn't back out the updated client.go after talking to Ryan. It's benign.